sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -110,6 +110,7 @@ object Cast {
     case (StringType, TimestampType | DateType) => true
     case (TimestampType | DateType, StringType) => true
     case (DateType, TimestampType) => true
+    case (DateType, StringType) => true
```

Comment:
Why the case is not handled by the case above?
```scala
case (TimestampType | DateType, StringType) => true
```

Reply:
Yea it's already covered by L111. We should close this PR.